src/medical_ui.cpp

```diff
@@ -0,0 +1,647 @@
+            mvwputch(window, point(pain_str_pos - 2, i), BORDER_COLOR, LINE_XOXO); // |
+        }
+        mvwputch(window, point(pain_str_pos - 2, 0), BORDER_COLOR, LINE_OXXX); // ^|^   
+        mvwputch(window, point(pain_str_pos - 2, 2), BORDER_COLOR, LINE_XXOX); // _|_
+        right_indent += utf8_width(remove_color_tags(pain_str)) + 3;
+    }
+
+    /* Hunger, Thirst & Fatigue Indicator */
+
+    const std::pair<std::string, nc_color> hunger_pair = display::hunger_text_color(you);
+    const std::pair<std::string, nc_color> thirst_pair = display::thirst_text_color(you);
+    const std::pair<std::string, nc_color> fatigue_pair = display::fatigue_text_color(you);
+    int cur_str_pos = 0;
+
+    // Hunger
+    if (hunger_pair.first.size() > 0) {
+        cur_str_pos = right_print(window, 1, right_indent, hunger_pair.second, hunger_pair.first);
+
+        /* ~~Borders */
+        for (int i = 1; i < getmaxy(window) - 1; i++) {
+            mvwputch(window, point(cur_str_pos - 2, i), BORDER_COLOR, LINE_XOXO); // |
+        }
+        mvwputch(window, point(cur_str_pos - 2, 0), BORDER_COLOR, LINE_OXXX); // ^|^   
+        mvwputch(window, point(cur_str_pos - 2, 2), BORDER_COLOR, LINE_XXOX); // _|_
+
+        right_indent += utf8_width(hunger_pair.first) + 3;
+    }
+
+    // Thirst
+    if (thirst_pair.first.size() > 0) {
+        cur_str_pos = right_print(window, 1, right_indent, thirst_pair.second, thirst_pair.first);
+
+        /* ~~Borders */
+        for (int i = 1; i < getmaxy(window) - 1; i++) {
+            mvwputch(window, point(cur_str_pos - 2, i), BORDER_COLOR, LINE_XOXO); // |
+        }
+        mvwputch(window, point(cur_str_pos - 2, 0), BORDER_COLOR, LINE_OXXX); // ^|^   
+        mvwputch(window, point(cur_str_pos - 2, 2), BORDER_COLOR, LINE_XXOX); // _|_
+
+        right_indent += utf8_width(thirst_pair.first) + 3;
+    }
+
+    //Fatigue
+    if (fatigue_pair.first.size() > 0) {
+        cur_str_pos = right_print(window, 1, right_indent, fatigue_pair.second, fatigue_pair.first);
+
+        /* ~~Borders */
+        for (int i = 1; i < getmaxy(window) - 1; i++) {
+            mvwputch(window, point(cur_str_pos - 2, i), BORDER_COLOR, LINE_XOXO); // |
+        }
+        mvwputch(window, point(cur_str_pos - 2, 0), BORDER_COLOR, LINE_OXXX); // ^|^   
+        mvwputch(window, point(cur_str_pos - 2, 2), BORDER_COLOR, LINE_XXOX); // _|_
+
+        right_indent += utf8_width(fatigue_pair.first) + 3;
+    }
+
+    /* Hotkey Helper */
+    std::string desc;
+    desc = string_format(_(
+        "[<color_yellow>%s</color>]Apply Item [<color_yellow>%s</color>]Keybindings"),
+        ctxt.get_desc("APPLY"), ctxt.get_desc("HELP_KEYBINDINGS"));
+
+    const int max_width = right_indent + TAB_WIDTH;
+    if (WIDTH - max_width > 3) // If the window runs out of room, we won't print keybindings.
+    {
+        right_print(window, 1, right_indent, c_white, desc);
+    }
+}
+
+void avatar::disp_medical()
```

Comment:
This has almost 500 lines. Would be great if you could split it up a bit to increase readability and maintainability.

Reply:
Done; Each column is it's own function & class in the hopes that the columns can eventually be built as JSON widgets with little effort.